### PR TITLE
Feature/submarine security spark 3.2

### PR DESCRIPTION
**Pull request by RamakrishnaChilaka**
### What is this PR for?
This pr supports spark 3.2

### What type of PR is it?
[Bug Fix | Improvement | Feature | Documentation | Hot Fix | Refactoring]
Improvement, we are giving spark 3.2 support

### Todos
* [ ] - Task

### Questions:
* Do the license files need updating? Yes/No
* Are there breaking changes for older versions? Yes/No
* Does this need new documentation? Yes/No

### CODE REVIEW

File: submarine-security/spark-security/pom.xml [hunk 1]
```diff
@@ -572,6 +572,20 @@
   </build>
 
   <profiles>
+    <profile>
+      <id>spark-3.2</id>
+      <properties>
+        <spark.version>3.2.0</spark.version>
+        <scala.version>2.12.14</scala.version>
```
Comment (akizminet) [inline submarine-security/spark-security/pom.xml:579]: why scala.version here is 2.12.14 instead of 2.12.15

Comment (anumee) [reply submarine-security/spark-security/pom.xml:579]: I agree with @akizminet. This should be 2.12.15 to match line #51.

File: submarine-security/spark-security/pom.xml [hunk 2]
```diff
@@ -572,6 +572,20 @@
...
+        <scala.binary.version>2.12</scala.binary.version>
+        <!--<scalatest.version>3.2.0</scalatest.version>-->
+        <spark.compatible.version>3</spark.compatible.version>
+        <commons-lang3.version>3.10</commons-lang3.version>
+        <jackson-databind.version>2.10.5</jackson-databind.version>
+        <jackson-annotations.version>2.10.5</jackson-annotations.version>
```
Comment (anumee) [inline submarine-security/spark-security/pom.xml:585]: Can we also add following here please?
`<ranger.version>2.1.0</ranger.version>`
`<ranger.major.version>2.1</ranger.major.version>`
`<httpcomponents.httpclient.version>4.5.13</httpcomponents.httpclient.version>`
`<httpcomponents.httpcore.version>4.4.15</httpcomponents.httpcore.version>`
`<httpcomponents.httpmime.version>4.5.13</httpcomponents.httpmime.version>`

**Review by anumee — COMMENTED** [review]
Thank you for starting this PR. Exactly what I am looking for!

File: submarine-security/spark-security/pom.xml (hunk 2 above)
Comment (RamakrishnaChilaka) [reply submarine-security/spark-security/pom.xml:585]: i agree @anumee , but ranger versions do have a seperate profile altogether.. as this spark 3.2 profile can be used with a different version of ranger as well.

File: submarine-security/spark-security/pom.xml [hunk 3]
```diff
@@ -363,7 +363,7 @@
       <plugin>
         <groupId>net.alchim31.maven</groupId>
         <artifactId>scala-maven-plugin</artifactId>
-        <version>3.2.2</version>
+        <version>4.5.4</version>
```
[the remark this answers is not in the document]
Comment (RamakrishnaChilaka) [reply submarine-security/spark-security/pom.xml]: Done

File: submarine-security/spark-security/pom.xml (hunk 1 above)
Comment (RamakrishnaChilaka) [reply submarine-security/spark-security/pom.xml]: Done

**Comment by xunliu** [comment]
hi @RamakrishnaChilaka , Security spark has been moved to the `apache/incubator-kyuubi` project, the `submarine` project will not contain the code for `security spark`. So this PR will not be merged.

**Pull request closed without merging**